WalletWasabi/WabiSabi/Client/CoinJoinManager.cs

```diff
@@ -90,24 +90,31 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
 					trackedWallets.Remove(finishedCoinJoin.Wallet.WalletName);
 					finishedCoinJoin.CancellationTokenSource.Dispose();
 
-					var finishedCoinJoinTask = finishedCoinJoin.CoinJoinTask;
-					if (finishedCoinJoinTask.IsCompletedSuccessfully)
+					try
 					{
-						Logger.LogInfo("Coinjoin client finished successfully!");
-					}
-					else if (finishedCoinJoinTask.IsFaulted)
-					{
-						if (finishedCoinJoinTask.Exception?.InnerException is InvalidOperationException ioe)
+						var success = await finishedCoinJoin.CoinJoinTask.ConfigureAwait(false);
```

Comment:
I wonder: Is it only possible to process `finishedCoinJoins` in serial fashion, or would it make sense to wrap `try-catch` in `Task.Run()`s and await `Task.WhenAll()`?

Reply:
Those finishedCoinJoins tasks are already completed, that means all the processing have been already done ant they are just a record in memory. For that reason `Task.WhenAll()` would return immediately. I think it is okay as it is. Anyway, if you find a better way just feel free to suggest it here.